### DISCUSSION TITLE
Many XRuns and ERRORS in Ubuntu Studio 22.04

### DISCUSSION THREAD

**Discussion by BeFrankly** [DISTRHO/Cardinal#376] [General]
Hello,
I started using Cardinal some month ago on my hardware that runs Ubuntu Studio 20.04 since today. Today I did a release upgrade and now the same hardware runs with Ubuntu Studio 22.04 (Jammy).
Jack (starting with QJackCtl) runs fine with the realtime kernel, even when I connect Ardour6 to it. Also Hydrogen drum machine runs fine and also did Carla.
But when I start Cardinal and attach it to Jack I get a lot of error messages in Jack, even when Cardinal is not playing any sound. Also there are a lot of xruns. Any idea?
uname -a
Linux Machine1 5.15.0-47-lowlatency #53-Ubuntu SMP PREEMPT Fri Aug 12 15:27:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

Here is the output of the jack message window:

Sat Sep 10 21:03:00 2022: New client 'Cardinal' with PID 23088
21:03:22.871 XRUN callback (1).
21:03:23.258 XRUN callback (1 übersprungen).
Sat Sep 10 21:03:22 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:22 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Sep 10 21:03:22 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:22 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Sep 10 21:03:23 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:23 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Sep 10 21:03:23 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:23 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
21:03:25.265 XRUN callback (5 übersprungen).
Sat Sep 10 21:03:24 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:24 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Sep 10 21:03:24 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:24 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Sep 10 21:03:25 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:25 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Sep 10 21:03:25 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:25 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error
Sat Sep 10 21:03:25 2022: ERROR: JackEngine::XRun: client = Cardinal was not finished, state = Running
Sat Sep 10 21:03:25 2022: ERROR: JackAudioDriver::ProcessGraphAsyncMaster: Process error

**Comment by falkTX** [comment]
this is not an issue with cardinal per se, xruns can happen if jack clients take too long to process.
I will convert this into a discussion it really doesnt belong as an issue ticket.

**Reply by BeFrankly** [reply]
Dear @falkTX , I know what you mean. But before the release upgrade it worked fine. And I have tried with different Frames and Period settings in Jack up to 2048/4 but still got this xruns. But I also found out that it does not happen if I just only use a clock, sequencer, oszillator, adsr and vca. So maybe Ubuntu changed the way audio is processed at the hardware side and it's not a Cardinal issue.

It's a shame. I just want to make music and not bugtracking. :-(

**Reply by falkTX** [reply]
nothing to do with cardinal then, if that is really the case. you can download earlier cardinal builds and compare

**Reply by BeFrankly** [reply]
Dear @falkTX , it's really crasy. I found out that it's the Liquid filter from Audible Instruments that let it happen. It's a filter I often use.
I started a fresh new project in Cardinal, added a sequencer, drum unit, oscillators and the Plateau Effect and no xruns happen. But even if I just add the Liquid filter (no need to connect cables to it) the xruns arrive in Jack. Even if no sound is playing and the clock for the sequencer is stopped. So what do you think about that. If I delete the Liquid filter, the xruns stopped immediately. That's a lot different from the behavior in Ubuntu 20.04.
I tested another filter, Holt from Rackwindows, and no xruns.

???

**Reply by BeFrankly** [reply]
Hello,
after using ubuntu 22.04 since three weeks now I can say, that this is really no Cardinal issue. There is a problem in initiating the usb device after waking up the PC and sometimes after the fist plugging in of the usb cable. Do not know exactly why this happens at the moment but definately no Cardinal issue.
